cmd/syncthing/gui.go

```diff
@@ -652,10 +652,14 @@ func restPostUpgrade(w http.ResponseWriter, r *http.Request) {
 func restPostScan(m *model.Model, w http.ResponseWriter, r *http.Request) {
 	qs := r.URL.Query()
 	folder := qs.Get("folder")
-	sub := qs.Get("sub")
-	err := m.ScanFolderSub(folder, sub)
-	if err != nil {
-		http.Error(w, err.Error(), 500)
+	if folder != "undefined" {
```

Comment:
I'd rather have the JS to explicitly exclude folder argument, this way no folder argument = scan everything, because from API perspective the semantic of `/rest/rescan?folder=undefined` is not great if someone were to write some external tool.